src/dns/coredns.rs

```diff
@@ -87,6 +87,7 @@ impl CoreDns {
                 },
                 v = receiver.next() => {
                     let msg_received = match v {
+                        Some(Err(err)) if err.kind() == std::io::ErrorKind::BrokenPipe => {break},
```

Comment:
This doesn't look right as you break the loop for tcp and udp and there is no logic to "retry" (open the socket again). At the very least it would need to be removed from the thread_handles map so a new container spawn would allow us to add it again.

And we likely should check what kind of errors we can get here and which of those or fatal and which are recoverable in general when reading from the socket. EPIPE just doesn't make any sense to me, it is not documented in recvmsg which should be the underlying syscall (or does the tokio abstraction do anything weird)

Can you attach strace to it and provide the output of when this happens?

Reply:
I will attempt to trigger it again and see what I can do. I have attached a test as well that demonstrates the hang when a broken pipe occurs using just a simple facade.

Reply:
Without understanding why this happens it doesn't make much sense to just merge "random" fixes and hope the best. I don't see what generates EPIPE here so I like to see a strace at least to understand which syscall is returning that. I agree there might be fatal errors that should cause us to abort but this is not specific to this one I presume and not specific to udp either?

And if we do that then we must update the thread_handles correctly and I guess try to bind the socket again.

Reply:
I attached strace on my coreOS instance using a rootful toolbox container and have managed to trigger the issue again but rather stupidly did not trace child threads / processes. Trying again now.